simple_app/plots.py

```diff
@@ -48,18 +85,37 @@ def specplot(query: str, db_file: str, nightskytheme: Theme) -> Tuple[Optional[s
     failstr: str
         the failed spectra
     """
-    def normalise(fluxarr: np.ndarray) -> np.ndarray:
-        fluxarr = (fluxarr - np.nanmin(fluxarr)) /\
-                  (np.nanmax(fluxarr) - np.nanmin(fluxarr))
-        return fluxarr
+    def normalise() -> np.ndarray:
+        wavestart = wave[0]
+        waveend = wave[-1]
+        objminwave, objmaxwave = minwave, maxwave = 0.81, 0.82
+        if wavestart > minwave and waveend > minwave and wavestart > maxwave and waveend > maxwave:
+            objminwave = wavestart
+            objmaxwave = wavestart + 0.01
+        elif minwave < wavestart < maxwave <= waveend and waveend > minwave:
+            objminwave = wavestart
+            objmaxwave = maxwave
+        elif wavestart <= minwave < waveend and wavestart < maxwave <= waveend:
+            objminwave = minwave
+            objmaxwave = maxwave
+        elif wavestart <= minwave < waveend < maxwave and wavestart < maxwave:
+            objminwave = minwave
+            objmaxwave = waveend
+        elif wavestart < minwave and waveend < minwave and wavestart < maxwave and waveend < maxwave:
+            objminwave = waveend - 0.01
+            objmaxwave = waveend
```

Comment:
Is it possible to collapse this logic somewhat? Maybe using something like numpy.clip? You may still need some if/else statements but maybe it can be made more readable.

Reply:
Yes thankyou, that works perfectly. The javascript version has to remain a mess to account for all the cases but the init case for python can be simplified massively.
There's probably a better way of doing it in javascript but this is fast and works (and I don't know that much javascript...)